envoy/network/proxy_protocol.h

```diff
@@ -26,5 +26,18 @@ struct ProxyProtocolData {
   }
 };
 
+enum class ProxyProtocolVersion {
+  // The proxy-protocol filter is not present in the filter chain and another filter has set
+  // the proxy-protocol version.
+  NotUsed = 0,
+  // The proxy-protocol filter did not find any proxy-protocol header and was skipped.
+  NotFound = 1,
+  V1 = 2,
+  V2 = 3
+};
+
+struct ProxyProtocolDataWithVersion : public ProxyProtocolData {
+  const ProxyProtocolVersion version_;
```

Comment:
You can avoid needing a separate enum with `NotUsed` by making this an `absl::optional`. Then you don't need to cast between enum types.